packages/e2e/run-tests.js

```diff
@@ -16,26 +15,25 @@ if (process.argv[2] !== "npm" && process.argv[2] !== "yarn") {
 const isYarn = process.argv[2] === "yarn";
 
 async function main() {
-  const fixtures = fsExtra
-    .readdirSync(fixtureProjectsDir)
-    .map((fixture) => path.join(fixtureProjectsDir, fixture));
-
-  // we run the cleanup both before and after running the tests, just in case
-  cleanup(fixtures);
-
-  // build hardhat and install it in each fixture project
-  const hardhatPackagePath = setup(fixtures);
+  // build hardhat and geth the path to the tgz
+  const hardhatPackagePath = buildHardhat();
   shell.cd(__dirname);
 
   // we don't throw if the tests fail so that we can cleanup things properly
   shell.set("+e");
-  const mochaResult = shell.exec(`mocha --recursive \"test/**/*.ts\"`);
+  const mochaResult = shell.exec(`mocha --recursive \"test/**/*.ts\"`, {
+    env: {
+      ...process.env,
+      // the tests need this information to setup the fixture projects
+      HARDHAT_E2E_PATH_TO_HARDHAT_TGZ: hardhatPackagePath,
+      HARDHAT_E2E_IS_YARN: isYarn,
+    },
+  });
   shell.set("-e");
 
-  cleanup(fixtures);
-  // we remove the tgz file because some tests might reinstall the fixture
-  // project dependencies, and that could fail if the tgz file isn't there
-  // anymore
+  // we remove the tgz file after the tests are run, because some of them might
+  // reinstall the fixture project dependencies, and that would fail if the tgz
+  // file isn't there anymore
```

Comment:
This comment is confusing to me.

"after the tests are run" ... as opposed to what?

And, it's saying, "we remove the tgz file...because...[some things] would fail if the tgz file isn't there anymore" ... seems counterintuitive.

Reply:
Thanks, this was indeed not very useful. I updated the comment. Does it look better now?

Reply:
It does.  But, it makes me wonder if that's the right thing to do.  When the tests fail and need investigation, it's super nice to have that .tgz around so that I can easily replicate the failure.  And re-acquainting myself with the code enough to find that line and comment it out feels pretty inefficient.  Would it be possible/practical to do any of:
* leave the file behind in the event of failure,
* store the file in /tmp/whatever instead of the package directory so that we can leave it behind all the time without causing too much noise, or
* do both: put it in /tmp, leave it in the case of failure, delete it in the case of success.

(To be clear, my Approval isn't contingent on answering these questions, much less implementing the decision.)

Reply:
That makes sense, but for now I guess we can just comment out that line if we need to keep the tgz file. This approach wouldn't be enough if the tests fail non-deterministically (so we can't just re-run them and comment out that line) **and** the tgz is important to understand why it failed, which seems very unlikely to me.

(Merging the PR, but we can continue discussing this anyway!)